code/modules/clothing/spacesuits/plasmamen.dm

```diff
@@ -118,33 +118,45 @@
 	. = ..()
 	if(!up)
 		. += visor_icon
-
-/obj/item/clothing/head/helmet/space/plasmaman/attackby(obj/item/hitting_item, mob/living/user)
-	. = ..()
-	if(istype(hitting_item, /obj/item/toy/crayon))
-		if(smile == FALSE)
-			var/obj/item/toy/crayon/CR = hitting_item
-			to_chat(user, span_notice("You start drawing a smiley face on the helmet's visor.."))
-			if(do_after(user, 2.5 SECONDS, target = src))
-				smile = TRUE
-				smile_color = CR.paint_color
-				to_chat(user, "You draw a smiley on the helmet visor.")
-				update_appearance()
-		else
+	if(smile)
+		var/mutable_appearance/slimey = mutable_appearance(icon, smile_state)
```

Comment:
smiley you mean?

Reply:
Also why make a new one every update appearance instead of doing what ``visor_icon`` does?

Reply:
potato told me a while back that caching these isn't a great idea, its not called nearly often enough to justify that